### PR TITLE
'updated' filter not working #330

### LINKED ISSUE #330: 'updated' filter not working
Here's the log:

```
648125 07 Jan 2015 18:23:58 ERROR UI [JavaFX Application Thread] - null
java.lang.NullPointerException
	at filter.expression.Qualifier.satisfiesUpdatedHours(Qualifier.java:313) ~[HubTurbo.jar:?]
	at filter.expression.Qualifier.isSatisfiedBy(Qualifier.java:131) ~[HubTurbo.jar:?]
	at filter.expression.Qualifier.process(Qualifier.java:82) ~[HubTurbo.jar:?]
	at ui.issuecolumn.IssueColumn.lambda$10(IssueColumn.java:242) ~[HubTurbo.jar:?]
	at ui.issuecolumn.IssueColumn$$Lambda$507/426882296.test(Unknown Source) ~[?:?]
	at javafx.collections.transformation.FilteredList.refilter(Unknown Source) ~[jfxrt.jar:?]
	at javafx.collections.transformation.FilteredList.access$100(Unknown Source) ~[jfxrt.jar:?]
	at javafx.collections.transformation.FilteredList$1.invalidated(Unknown Source) ~[jfxrt.jar:?]
	at javafx.beans.property.ObjectPropertyBase.markInvalid(Unknown Source) ~[jfxrt.jar:?]
	at javafx.beans.property.ObjectPropertyBase.set(Unknown Source) ~[jfxrt.jar:?]
	at javafx.collections.transformation.FilteredList.<init>(Unknown Source) ~[jfxrt.jar:?]
	at ui.issuecolumn.IssueColumn.refreshItems(IssueColumn.java:307) ~[HubTurbo.jar:?]
	at ui.issuepanel.IssuePanel.refreshItems(IssuePanel.java:47) ~[HubTurbo.jar:?]
	at ui.issuecolumn.IssueColumn.applyFilterExpression(IssueColumn.java:243) ~[HubTurbo.jar:?]
	at ui.issuecolumn.IssueColumn.applyStringFilter(IssueColumn.java:225) ~[HubTurbo.jar:?]
	at ui.issuecolumn.IssueColumn.lambda$3(IssueColumn.java:96) ~[HubTurbo.jar:?]
	at ui.issuecolumn.IssueColumn$$Lambda$492/1222681631.apply(Unknown Source) ~[?:?]
	at ui.components.FilterTextField.confirmEdit(FilterTextField.java:221) ~[HubTurbo.jar:?]
	at ui.components.FilterTextField.lambda$6(FilterTextField.java:94) ~[HubTurbo.jar:?]
	at ui.components.FilterTextField$$Lambda$491/105150517.handle(Unknown Source) ~[?:?]
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.event.EventUtil.fireEvent(Unknown Source) ~[jfxrt.jar:?]
	at javafx.event.Event.fireEvent(Unknown Source) ~[jfxrt.jar:?]
	at javafx.scene.Scene$KeyHandler.process(Unknown Source) ~[jfxrt.jar:?]
	at javafx.scene.Scene$KeyHandler.access$1800(Unknown Source) ~[jfxrt.jar:?]
	at javafx.scene.Scene.impl_processKeyEvent(Unknown Source) ~[jfxrt.jar:?]
	at javafx.scene.Scene$ScenePeerListener.keyEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(Unknown Source) ~[jfxrt.jar:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_20]
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.glass.ui.View.handleKeyEvent(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.glass.ui.View.notifyKey(Unknown Source) ~[jfxrt.jar:?]
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method) [jfxrt.jar:?]
	at com.sun.glass.ui.win.WinApplication.lambda$null$141(Unknown Source) [jfxrt.jar:?]
	at com.sun.glass.ui.win.WinApplication$$Lambda$37/626202354.run(Unknown Source) [jfxrt.jar:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_20]
```

### PR DESCRIPTION
**Pull request by dariusf**
Fixes #330

### CODE REVIEW

**Comment by damithc** [comment]
Can add a test case to cover this case.

**Comment by dariusf** [comment]
Currently there are no tests for issues satisfying filters -- the tests are only for parsing. I've set up the basic framework. Will add more after #338 is merged.

**Comment by dariusf** [comment]
Added tests covering all filters.

File: src/tests/EvalTests.java [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+		assertEquals(Qualifier.process(Parser.parse("author:BOB"), issue), true);
+		assertEquals(Qualifier.process(Parser.parse("author:bob"), issue), true);
+		assertEquals(Qualifier.process(Parser.parse("author:alice"), issue), false);
+		assertEquals(Qualifier.process(Parser.parse("author:o"), issue), true);
+		assertEquals(Qualifier.process(Parser.parse("author:lic"), issue), false);
+	}
+
+	@Test
+	public void involves() {
+		// Succeeds if author and assignee succeed
```
Comment (damithc) [inline src/tests/EvalTests.java]: To be done in future? in that case, add a `//TODO` comment (tracked by eclipse)

**Pull request merged**